Userland/Applications/Assistant/Providers.cpp

```diff
@@ -165,6 +165,9 @@ void FileProvider::build_filesystem_cache()
 
     (void)Threading::BackgroundAction<int>::construct(
         [this](auto&) {
+            RefPtr<FileProvider> guard = this;
+            (void)guard;
```

Comment:
How about this?
```suggestion
        [this, strong_ref = NonnullRefPtr(*this)](auto&) {
```

Reply:
Huh, didn't know that was a thing you could do. Sure.